Automattic/wp-calypso
# Newsletters: add paid Newsletter to post-setup step

Follow up to https://github.com/Automattic/wp-calypso/pull/76042

## Proposed Changes

* Move "paid newsletters" checkbox into it's own component. I (@edanzer) updated this PR so that ALL logic related to the component is now within the component. This makes it simpler to re-use. There is no need for extra logic in the setup screen, or in the useSetupFormInitialValues hook. 
* Use the new component in both "newsletter setup" and "newsletter post setup" steps. Latter step is shown when returning to newsletter setup from the launchpad for already created site.

## Testing Instructions

* Launch a new Newsletter site at http://calypso.localhost:3000/setup/newsletter/intro
* On first Set Up page, confirm you can check/uncheck the box as expected. Check it and continue to Launchpad. 
* From Launchpad, click Personalize newsletter to go to the Newsletter Post Setup page. 
* Confirm that the checkbox is checked when this page loads (assuming you checked it on the first screen)
* Try checking/unchecking this box, returning to Launchpad, and coming back. Confirm that the status of the checkbox is alway accurate and what you expect.

## Pre-merge Checklist
